### ISSUE TITLE
Cannot bind read-only array to certain control properties

### ISSUE THREAD

**Issue by taco-indenbosch** [primefaces/primeng#9411]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => Search github for a similar issue or PR before submitting
[x] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

There are some properties of certain controls that accept an array of values. For example the suggestions property of the autocomplete control is defined as "any[]" (in the class Autocomplete in autocomplete.d.ts). I assume that the control does not modify these values. However, I cannot bind this property to a readonly array as it is not defined as "readonly any[]". Of course I can create a copy of the array before passing it, but that is inefficient and can lead to problems.

In other words, I think that array valued properties such as this should be defined as "readonly any[]".

* **Angular version:** 10.1.4

* **PrimeNG version:** 10.0.3

* **Language:** TypeScript 3.9

* **Node (for AoT issues):** `node --version` =  12.8.14

**Comment by yigitfindikli** [comment]
Issue is not clear, please feel free to reopen with a stackblitz example.
